packages/core/src/services/documents/modify.ts

```diff
@@ -0,0 +1,136 @@
+import { omit } from 'lodash-es'
+
+import { readMetadata } from '@latitude-data/compiler'
+import {
+  DocumentVersion,
+  documentVersions,
+  getDocumentsAtCommit,
+  Result,
+  Transaction,
+} from '@latitude-data/core'
+import { LatitudeError, NotFoundError } from '$core/lib/errors'
+
+import { existsAnotherDocumentWithSamePath, getEditableCommit } from './utils'
+
+async function findDocumentsWithUpdatedHash(documents: DocumentVersion[]) {
+  const getDocumentContent = async (path: string): Promise<string> => {
+    const document = documents.find((d) => d.path === path)
+    if (!document) {
+      throw new Error(`Document not found`)
+    }
+    return document.content
+  }
+
+  const updatedDocuments: DocumentVersion[] = []
+  for (const document of documents) {
+    const { hash: newHash } = await readMetadata({
+      prompt: document.content ?? '',
+      referenceFn: getDocumentContent,
+    })
+
+    if (newHash !== document.hash) {
+      updatedDocuments.push({
+        ...document,
+        hash: newHash,
+      })
+    }
+  }
+
+  return updatedDocuments
+}
+
+export async function modifyExistingDocument({
+  commitId,
+  documentUuid,
+  path,
+  content,
+  deletedAt,
+}: {
+  commitId: number
+  documentUuid: string
+  path?: string
+  content?: string | null
+  deletedAt?: Date | null
+}) {
+  const commitResult = await getEditableCommit(commitId)
+  if (commitResult.error) return commitResult
+
+  const currentDocuments = await getDocumentsAtCommit({
+    commitId,
+    commitMergedAt: commitResult.value.mergedAt,
+  })
+  if (currentDocuments.error) return currentDocuments
```

Comment:
hmmm we this result error check in a million places already I wonder if we can improve. How about a try catch for this whole method and you just wrap whatever error you get in a Result.error?